lib/SIL/SILBasicBlock.cpp

```diff
@@ -29,12 +29,15 @@ using namespace swift;
 // SILBasicBlock Implementation
 //===----------------------------------------------------------------------===//
 
-SILBasicBlock::SILBasicBlock(SILFunction *parent, SILBasicBlock *afterBB)
+SILBasicBlock::SILBasicBlock(SILFunction *parent, SILBasicBlock *relativeToBB,
+                             bool after)
     : Parent(parent), PredList(nullptr) {
-  if (afterBB) {
-    parent->getBlocks().insertAfter(afterBB->getIterator(), this);
-  } else {
+  if (!relativeToBB) {
     parent->getBlocks().push_back(this);
+  } else if (after) {
```

Comment:
Early returns instead of chained else-if?

Reply:
Good point.  On balance, I think the chained `else-if` is probably better in this case given the length of the cases (all very short, so adding the `return` would use more space) and the fact that we're in a constructor (and therefore might very plausibly need to add more code after this to handle future state).